com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ColladaUtil.java

```diff
@@ -284,34 +281,89 @@ private static Matrix4d getBoneLocalToParent(Bone bone) {
-            animBuilder.addTracks(animTrackBuilder.build());
-        }
+    private static void samplePosTrack(Rig.RigAnimation.Builder animBuilder, Rig.AnimationTrack.Builder animTrackBuilder, RigUtil.AnimationTrack track, double duration, double startTime, double sampleRate, double spf, boolean interpolate) {
+        if (track.keys.isEmpty())
+            return;
+
+        RigUtil.PositionBuilder positionBuilder = new RigUtil.PositionBuilder(animTrackBuilder);
+        RigUtil.sampleTrack(track, positionBuilder, new Point3d(0.0, 0.0, 0.0), startTime, duration, sampleRate, spf, true);
+    }
+
+    private static void sampleRotTrack(Rig.RigAnimation.Builder animBuilder, Rig.AnimationTrack.Builder animTrackBuilder, RigUtil.AnimationTrack track, double duration, double startTime, double sampleRate, double spf, boolean interpolate) {
+        if (track.keys.isEmpty())
+            return;
+
+        RigUtil.QuatRotationBuilder rotationBuilder = new RigUtil.QuatRotationBuilder(animTrackBuilder);
+        RigUtil.sampleTrack(track, rotationBuilder, new Quat4d(0.0, 0.0, 0.0, 1.0), startTime, duration, sampleRate, spf, true);
     }
 
-    private static void sampleRotTrack(Rig.RigAnimation.Builder animBuilder, int boneIndex, RigUtil.AnimationTrack track, double duration, double startTime, double sampleRate, double spf, boolean interpolate) {
-        if (!track.keys.isEmpty()) {
-            Rig.AnimationTrack.Builder animTrackBuilder = Rig.AnimationTrack.newBuilder();
-            animTrackBuilder.setBoneIndex(boneIndex);
-            RigUtil.QuatRotationBuilder rotationBuilder = new RigUtil.QuatRotationBuilder(animTrackBuilder);
-            RigUtil.sampleTrack(track, rotationBuilder, new Quat4d(0.0, 0.0, 0.0, 1.0), startTime, duration, sampleRate, spf, true);
-            animBuilder.addTracks(animTrackBuilder.build());
+    private static void sampleScaleTrack(Rig.RigAnimation.Builder animBuilder, Rig.AnimationTrack.Builder animTrackBuilder, RigUtil.AnimationTrack track, double duration, double startTime, double sampleRate, double spf, boolean interpolate) {
+        if (track.keys.isEmpty())
+            return;
+
+        RigUtil.ScaleBuilder scaleBuilder = new RigUtil.ScaleBuilder(animTrackBuilder);
+        RigUtil.sampleTrack(track, scaleBuilder, new Vector3d(1.0, 1.0, 1.0), startTime, duration, sampleRate, spf, true);
+    }
+
+    public static void createAnimationTracks(Rig.RigAnimation.Builder animBuilder,
+                                             RigUtil.AnimationTrack posTrack,
+                                             RigUtil.AnimationTrack rotTrack,
+                                             RigUtil.AnimationTrack sclTrack,
+                                            int boneIndex, double duration, double startTime, double sampleRate) {
+        double spf = 1.0 / sampleRate;
+
+        Rig.AnimationTrack.Builder animTrackBuilder = Rig.AnimationTrack.newBuilder();
+        animTrackBuilder.setBoneIndex(boneIndex);
```

Comment:
We now create one track for each bone, whereas earlier we created one track per bone and pos/rot/scl